python2-pybind11/challenge/basics.cpp

```diff
@@ -0,0 +1,40 @@
+#include <pybind11/pybind11.h>
+
+#include "basics.hpp"
+
+namespace py = pybind11;
+
+PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>);
+
+namespace basics {
+
+PYBIND11_PLUGIN(basics) {
```

Comment:
Yes, that does seem to be the case. Actually ``PYBIND11_PLUGIN`` is not ``extern "C"`` but it just does a bunch of function calls and then calls another macro ``PYBIND11_PLUGIN_IMPL`` which is ``extern "C"``. But for clarity (and to prevent name clashes when the same module is defined in different namespaces) I moved it out.